helm/dex-app/templates/dex-secret.yaml

```diff
@@ -0,0 +1,40 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+{{ include "dex.labels" . | indent 4 }}
+    app.kubernetes.io/component: dex
+  name: dex
+stringData:
+  config.yaml: |-
+    web:
+      http: 0.0.0.0:5556
+    storage:
+      type: kubernetes
+      config:
+        inCluster: true
+    logger:
+      level: debug
+    oauth2:
+      responseTypes: ["code", "token", "id_token"]
+      alwaysShowLoginScreen: false
+      skipApprovalScreen: true
+    enablePasswordDB: false
+    frontend:
+      logoURL: http://styleguide.io/o/giantswarm/logo_simplified/giantswarm_logo_simplified.png
+    issuer: https://{{ .Values.Installation.V1.GiantSwarm.OIDC.IssuerAddress }}
+    expiry:
+      signingKeys: "{{ .Values.Installation.V1.GiantSwarm.OIDC.Expiry.SigningKeys }}"
+      idTokens: "{{ .Values.Installation.V1.GiantSwarm.OIDC.Expiry.IdTokens }}"
+    staticClients:
+    - id: dex-k8s-authenticator
+      redirectURIs:
+      - https://{{ .Values.Installation.V1.GiantSwarm.OIDC.ClientAddress }}/callback
+      name: dex-k8s-authenticator
+      secret: {{ .Values.Installation.V1.Secret.OIDC.ClientSecret }}
+    connectors:
+    - type: {{ .Values.Installation.V1.GiantSwarm.OIDC.Type }}
```

Comment:
then you can iterate over IdPs to fill connectors section

Reply:
I'm pretty sure it will be easier with this. But for now, it is all designed to have single OIDC provider configured. When we will get to the point where there is requirement about multiple IdPs - we can add that later. 
I just don't want to introduce too many changes at once. I'm already changing so many things outside of the context